src/observation/encodeMatcher/index.js

```diff
@@ -59,17 +60,10 @@ const encodeConditions: (Where[]) => Matcher<*> = pipe(
 )
 
 export default function encodeMatcher<Element: Model>(query: QueryDescription): Matcher<Element> {
-  const { joinTables, nestedJoinTables, sortBy, take, skip, lokiFilter, where } = query
-
   invariant(
-    !joinTables.length &&
-      !nestedJoinTables.length &&
-      !sortBy.length &&
-      !take &&
-      !skip &&
-      !lokiFilter,
+    canEncodeMatcher(query),
     `Queries with joins, sortBy, take, or skip can't be encoded into a matcher`,
```

Comment:
Hmm now it's going to be even harder to keep this comment up to date. Maybe add `forbiddenClauses` array to `canEncode.js` and here `join` into string?

Reply:
good point, i'll do that

Reply:
fixed in next PR